app/waf/src/controllers/adc.controller.ts

```diff
@@ -253,14 +235,23 @@ export class AdcController extends BaseController {
     try {
       await this.serialize(adc, {status: AdcState.PARTITIONING, lastErr: ''});
       await this.asgMgr.deploy(mgmt.ipAddress, mgmt.tcpPort, paritionObj);
-      await this.serialize(adc, {status: AdcState.ACTIVE, lastErr: ''});
+      await checkAndWait(
+        () => this.adcStCtr.gotTo(AdcState.PARTITIONED),
+        60,
+      ).then(
+        async () => this.serialize(adc, {status: AdcState.ACTIVE, lastErr: ''}),
+        async () =>
+          this.serialize(adc, {
+            status: AdcState.PARTITIONERR,
+            lastErr: `${AdcState.PARTITIONERR}: Fail to create partition`,
+          }),
+      );
     } catch (err) {
       this.logger.error(`Creating partition ${tenantName} Error.`);
       await this.serialize(adc, {
         status: AdcState.PARTITIONERR,
-        lastErr: `${AdcState.PARTITIONERR}: Fail to create partition`,
+        lastErr: `${AdcState.PARTITIONERR}: Fail to create partition: ${err}`,
```

Comment:
Will the err be converted to string automatically?

Reply:
Changed it. thx.

Reply:
Went through the changes and thanks for adding the partition related UT. :-)